src/Features/LanguageServer/Protocol/Handler/CodeLens/CodeLensResolveHandler.cs

```diff
@@ -38,20 +40,37 @@ public LSP.TextDocumentIdentifier GetTextDocumentIdentifier(LSP.CodeLens request
     public async Task<LSP.CodeLens> HandleRequestAsync(LSP.CodeLens request, RequestContext context, CancellationToken cancellationToken)
     {
         var document = context.GetRequiredDocument();
+        var currentDocumentSyntaxVersion = await document.GetSyntaxVersionAsync(cancellationToken).ConfigureAwait(false);
         var resolveData = GetCodeLensResolveData(request);
-        var (cacheEntry, memberToResolve) = GetCacheEntry(resolveData);
 
-        var currentSyntaxVersion = await document.GetSyntaxVersionAsync(cancellationToken).ConfigureAwait(false);
-        var cachedSyntaxVersion = cacheEntry.SyntaxVersion;
-
-        if (currentSyntaxVersion != cachedSyntaxVersion)
+        request.Command = new LSP.Command
         {
-            throw new LocalRpcException($"Cached resolve version {cachedSyntaxVersion} does not match current version {currentSyntaxVersion}")
+            Title = string.Format(FeaturesResources._0_references_unquoted, "-"),
+            CommandIdentifier = ClientReferencesCommand,
+            Arguments = new object[]
             {
-                ErrorCode = LspErrorCodes.ContentModified
-            };
+                resolveData.TextDocument.Uri,
+                request.Range.Start
+            }
+        };
+
+        // If the request is for an older document (e.g. older syntax version)
+        // return a request with empty information.
+        if (resolveData.ResultId != currentDocumentSyntaxVersion.ToString())
+        {
+            context.TraceInformation($"Requested syntax version {resolveData.ResultId} does not match current version {currentDocumentSyntaxVersion}");
+            return request;
         }
 
+        // If the cache has no information, return an empty request
+        var cacheEntry = _codeLensCache.GetCachedEntry(currentDocumentSyntaxVersion.ToString());
+        if (cacheEntry == null)
+        {
+            context.TraceInformation($"No cached information currently available for {currentDocumentSyntaxVersion}");
+            return request;
```

Comment:
Just tracing and returning an empty request when the cache is full isn't great. We were throwing an exception, but we should probably take the perf hit to re-calculate the information the cache is missing. It's not the callers fault we don't have the info cached anymore.

Reply:
Yup - verify in the above condition that the resolve request matches the current document syntax version, it should be totally fine to recalculate the members here.  In fact the recalculation of member locations shouldn't even really be expensive - its a syntax-only operation on the document.